Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationFailureRepositoryTest.cs

```diff
@@ -0,0 +1,440 @@
+                     };
+
+      repository.AddCellValidationFailures(_rowObjectStub.Object, _columnDefinitionStub.Object, failures);
+
+      var result = repository.GetValidationFailuresForDataCell(_rowObjectStub.Object, _columnDefinitionStub.Object, false);
+      Assert.That(result, Is.EqualTo(failures));
+    }
+
+    [Test]
+    public void ClearAllFailures_NothingContainedInRepository ()
+    {
+      var repository = new BocListValidationFailureRepository();
+
+      var cellFailure = BusinessObjectValidationFailure.Create("Cell failure");
+      var rowFailure = BusinessObjectValidationFailure.Create("Row failure");
+      var listFailure = BusinessObjectValidationFailure.Create("List failure");
+
+      repository.AddCellValidationFailures(_rowObjectStub.Object, _columnDefinitionStub.Object, new[] { cellFailure });
+      repository.AddRowValidationFailures(_rowObjectStub.Object, new[] { rowFailure });
+      repository.AddListValidationFailures(new[] { listFailure });
+
+      repository.ClearAllFailures();
+
+      var result = repository.GetValidationFailuresForBocListAndContainingDataRowsAndDataCells(false);
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void GetValidationFailuresForDataCell_ReturnsCellFailures ()
+    {
+      var repository = new BocListValidationFailureRepository();
+
+      var cellFailure = BusinessObjectValidationFailure.Create("Cell failure");
+      var rowFailure = BusinessObjectValidationFailure.Create("Row failure");
+      var listFailure = BusinessObjectValidationFailure.Create("List failure");
+
+      repository.AddCellValidationFailures(_rowObjectStub.Object, _columnDefinitionStub.Object, new[] { cellFailure });
+      repository.AddRowValidationFailures(_rowObjectStub.Object, new[] { rowFailure });
+      repository.AddListValidationFailures(new[] { listFailure });
+
+      var result = repository.GetValidationFailuresForDataCell(_rowObjectStub.Object, _columnDefinitionStub.Object, false);
+      Assert.That(result, Does.Contain(cellFailure));
+      Assert.That(result, Does.Not.Contain(rowFailure));
+      Assert.That(result, Does.Not.Contain(listFailure));
+    }
+
+    [Test]
+    public void GetValidationFailuresForDataCell_MarkAsHandled_NotContainedInRepositoryAfterwards ()
+    {
+      var repository = new BocListValidationFailureRepository();
+
+      var failures = new[]
+                     {
+                       BusinessObjectValidationFailure.Create("A failure")
+                     };
+
+      repository.AddCellValidationFailures(_rowObjectStub.Object, _columnDefinitionStub.Object, failures);
+
+      var result = repository.GetValidationFailuresForDataCell(_rowObjectStub.Object, _columnDefinitionStub.Object, true);
+      Assert.That(result, Is.EqualTo(failures));
+
+      result = repository.GetValidationFailuresForDataCell(_rowObjectStub.Object, _columnDefinitionStub.Object, false);
```

Comment:
From a semantic point of view, the most important operation is that more general calls also do not include this failure. I think we should add to the tests that we add a failure for a second column and then use the parent opertion (e.g. the row operation) and verify that only the unhandled results are returned.

This applies to the other APIs as well.